samples/xbee/demos/EndToEndAgricultureSample/main.py

```diff
@@ -1102,14 +1254,31 @@ def time_task():
       - Update the tank level based on the position of the valves.
       - Start the irrigation process.
     """
-    global current_time, rain, tank_level
+    global current_time, rain, rain_previous, tank_level
 
     while True:
         start_time = time.time()
 
         # Reset the rain at midnight.
         if (current_time + time_factor) // SECONDS_PER_DAY > 0:
+            # Create and configure the serial port settings
+            ser = serial.Serial()
```

Comment:
Doesn't this instantiation conflict with the port created in the ``read_microbit()`` method? Maybe we should declare the serial port as global and use it in both methods.